python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+        logger.debug("remote device is %s", device)
+        ctx = remote.cpu() if device == "cpu" else remote.gpu()
+
+        if profile:
+            logger.debug("creating runtime with profiling enabled")
+            module = debug_runtime.create(graph, remote_lib, ctx, dump_root="./prof")
+        else:
+            logger.debug("creating runtime with profiling disabled")
+            module = runtime.create(graph, remote_lib, ctx)
+
+        logger.debug("load params into the runtime module")
+        module.load_params(params)
+
+        shape_dict, dtype_dict = get_input_info(graph, params)
+        inputs_dict = make_inputs_dict(inputs, shape_dict, dtype_dict, fill_mode)
+
+        logger.debug("setting inputs to the module")
+        module.set_input(**inputs_dict)
+
+        # Run must be called explicitly if profiling
+        if profile:
+            logger.debug("running the module with profiling enabled")
+            module.run()
+
+        # create the module time evaluator (returns a function)
+        timer = module.module.time_evaluator("run", ctx, 1, repeat=repeat)
+        # call the evaluator function to invoke the module and save execution times
+        prof_result = timer()
+        # collect a list of execution times from the profiling results
+        times = prof_result.results
+
+        logger.debug("collecting the output tensors")
+        num_outputs = module.get_num_outputs()
+        outputs = {}
+        for i in range(num_outputs):
+            output_name = "output_{}".format(i)
+            outputs[output_name] = module.get_output(i).asnumpy()
+
+        return outputs, times
+
+
+def get_top_results(outputs, max_results):
+    """Return the top n results from the output tensor.
+
+    This function is primarily for image classification and will
+    not necessarily generalise.
+
+    Parameters
+    ----------
+    outputs : dict
+        Outputs dictionary - {output_name: np.array}.
+    max_results : int
+        Number of results to return
+
+    Returns
+    -------
+    top_results : np.array
+        Results array of shape (2, n).
+        The first row is the indices and the second is the values.
+
+    """
+    output = outputs["output_0"]
+    sorted_labels = output.argsort()[0][-max_results:][::-1]
+    output.sort()
+    sorted_values = output[0][-max_results:][::-1]
+    top_results = np.array([sorted_labels, sorted_values])
+    return top_results
+
+
+def format_times(times):
+    """Format the mean, max, min and std of the execution times.
+
+    Parameters
+    ----------
+    times : list
+        A list of execution times (in seconds).
+
+    Returns
+    -------
+    str
+        A formatted string containing the statistics.
+    """
+    # This has the effect of producing a small table that looks like:
+    #
+    # Execution time summary:
+    # mean (s)   max (s)    min (s)    std (s)
+    # 0.14310    0.16161    0.12933    0.01004
```

Comment:
Better to put this into the docstring as a note so that people can easily see it in the API reference without looking into the code.

Reply:
fixed